tensorflow_probability/python/distributions/laplace_test.py

```diff
@@ -340,5 +343,30 @@ def testLaplaceNonPositiveInitializationParamsRaises(self):
           loc=loc_v, scale=scale_v, validate_args=True)
       self.evaluate(laplace.mean())
 
+  @tfe.run_test_in_graph_and_eager_modes()
+  def testLaplaceLaplaceKL(self):
+    batch_size = 6
+    event_size = 3
+
+    a_loc = np.array([[0.5] * event_size] * batch_size, dtype=np.float32)
+    a_scale = np.array([[0.1] * event_size] * batch_size, dtype=np.float32)
+    b_loc = np.array([[0.4] * event_size] * batch_size, dtype=np.float32)
+    b_scale = np.array([[0.2] * event_size] * batch_size, dtype=np.float32)
+
+    a = tfd.Laplace(loc=a_loc, scale=a_scale)
+    b = tfd.Laplace(loc=b_loc, scale=b_scale)
+
+    distance = tf.abs(a_loc - b_loc)
+    ratio = a_scale / b_scale
+    true_kl = -tf.log(ratio) - 1 + distance / b_scale + ratio * tf.exp(- distance / a_scale)
```

Comment:
this is essentially re-writing the code under test in the test. maybe at least use numpy here? i think that's what we usually do elsewhere (others might chime in?)

Reply:
I don't have a strong feeling here since using numpy is basically equivalent to using TF here. Unfortunately we don't have a 'KL test harness' that we can test against, so testing this will inherently require writing a copy of the KL divergence function somewhere (I think the main things we want to test are that you recover zero when you pass in the same input, that batching / broadcasting works). I guess writing the function in numpy could potentially force writing a simpler thing (although in this case the code is pretty simple).

I guess another thing we can do is check that the sample KL is consistent with the analytic KL.

Reply:
Added the sample KL test.